pkg/network/node/egressip.go

```diff
@@ -266,44 +233,120 @@ func (eip *egressIPWatcher) updateNamespaceEgress(vnid uint32, egressIP string)
 
 	ns := eip.namespacesByVNID[vnid]
```

Comment:
Need to consider join and isolate project operations otherwise there will be some problems.
Example: 
1. ns1 with vnid v1 requested egrees ip e1
2. ns2 with vnid v2 requested egress ip e2
3. ns2 joined with ns1, now ns2 has vnid v1
4. eip.namespacesByVNID[v1] will now have requestedIP=e2 
5. ns2 got isolated
Now, v1 has only one namespace ns1 but the requested ip will be set as ep2 instead of ep1 which is incorrect.

Reply:
filed #18852